### DISCUSSION TITLE
Another mode of this app, or a separate app?

### DISCUSSION THREAD

**Discussion by getify** [getify/youperiod.app#49] [Ideas]
It just occurred to me, when pregnant many women use apps (maybe not period-tracker apps) to track various details of their pregnancy (dates/date ranges, symptoms, gestation progress, etc).

And given how this app is already trying to provide privacy to women/those with uteruses, for tracking periods when-not-pregnant, it seems like the same types of privacy and data protection should extend to when-pregnant usage. I think the case is pretty strong that similar *risks* exist for having such when-pregnant data used against your will, by "big tech" or by governments/police.

Up for discussion: do we think it would make sense to have both "modes" in this same app... or should that just be a separate app (with the same underlying/shared technical approach for UX and privacy)?

In favor, it means that you only need to manage one app, one passphrase, one set of processes for any sort of backups/syncs/etc. And you get a continuity of sorts, of your health-data tracking, from before pregnancy, during pregnancy, and after pregnancy. It'd also benefit from continuity of design and UX, etc.

OTOH, the name/branding do rather strongly suggest when-not-pregnant usage, and it might be hard to attract someone to start using such an app once pregnant if they hadn't already been using it prior. I don't think I want to rename/rebrand this app to broaden, as I think its name/brand focus is important to rapidly capture attention in the current heightened-risk climate.

What do you all think? And just to be transparent, women/those with uteruses should drive this discussion and decision making. I'm just opening it up.

**Comment by aditiranka27** [comment]
With what I have seen with period tracking apps is that the pregnancy feature is within the app (in some counted as a premium feature). 

I think it is best to have the  **_pregnant_** mode in the app itself because in the long run along with the **_period tracking_** features we can add on the  **_trying to get pregnant_** feature which can have some educational resources (e.g tips, diet/nutrition etc) and the **_pregnant mode_** feature which includes pre natal and other details as @getify mentioned above. 

Also having all these feature in the same app is kind of accessible too, I would not have to maintain two different apps.

I feel the name/branding should not be that of an issue for the pregnancy feature because it is kind of expected to have this feature with a period tracking app (at least that is what I have seen).

**Comment by aditiranka27** [comment]
Well I am commenting here but this might be a feature discussion as being discussed in #37  (I can move this comment, if needed).

Please excuse my lack of technical knowledge here:  

But there is one more feature which I have used on the _flo_ app which I find very helpful (I have used only for period tracking), I have logged various symptoms on the app about my mood, headaches, backaches etc etc. for over a period of time and after having enough of my data, the app starts sending prediction notifications about how I might feel today and in my case they really have been spot on especially regarding my headaches due to hormonal changes (I am kind of mentally prepareed then). 

With that being said, we aim to not save data but is there anyway we can implement this kind of ML in our app. I would really like to hear from the experts here. 

Also, if anybody has used logging symptoms while being pregnant, then also please comment.

**Reply by getify** [reply]
I love that you've brought this up! I have contemplated similar questions, assuming that one of the benefits of the more corporate apps in "femtech" was that they could do these big data things on their servers and start offering a bunch of intelligence in predictiveness.

Since privacy is our main goal, I don't think we should ever cross the line of "oh, but we actually anonymize your data and ship it off to a big data factory.... blah blah". I think that would really defeat our main distinguishing *feature*.

I'm certainly no data-scientist or expert in ML/AI, but I have done some research in the area. What I understand is, we absolutely can run some light ML even client-side only in the app. There's literally JS/WASM builds of well known ML libraries and they can run off-thread (web workers) in the background, and be somewhat remotely responsive (multiple seconds, not minutes or hours) more than we might have otherwise assumed. Some of them even use fancy tricks like tricking the WebGL engine into running some of the math "hardware accelerated" on the GPU.

However... the big problem is, as far as I can tell, that you have to get quite a lot of data to make a useful MODEL to run the ML against. If we took all the data from all users, and we had 5000 regular users, and we collected 1000 pieces of data from each... 5M data points *might* start to approach the size of information to create the first model. But since we only want to use *your* data, how do we get millions of data points from a single user?

I don't have any answers to how we get around this problem. I would absolutely love to make this also be "the most intelligent period (and pregnancy!) tracking app". But *if* we have to choose between that and data privacy/security, I think the answer for this app should be clear.

I'm super interested in input from anyone else who has ideas on how we can get around this problem!

**Reply by pedropcamellon** [reply]
It’s an awesome idea. Maybe we don’t need that great model, super accurate, maybe we can just use averages. But certainly, I think we should include some predictions.
